src/main/java/com/future/function/qa/api/auth/AuthAPI.java

```diff
@@ -0,0 +1,57 @@
+package com.future.function.qa.api.auth;
+
+import com.future.function.qa.api.BaseAPI;
+import com.future.function.qa.model.request.auth.AuthWebRequest;
+import io.restassured.http.Cookie;
+import io.restassured.response.Response;
+import io.restassured.specification.RequestSpecification;
+import net.thucydides.core.annotations.Step;
+
+public class AuthAPI extends BaseAPI {
+
+  @Step
+  public Response getLoginStatus(Cookie cookie) {
+
+    if (cookie == null) {
+      return getLoginStatusWithoutCookie();
```

Comment:
can you remove the if statement? when the cookie is null, is it throw nullpointer?

Reply:
Unfortunately, it does (how annoying). Hence the if block